src/services/modules/rest/impl/src/main/java/org/geoserver/geofence/services/rest/impl/RESTAdminRuleServiceImpl.java

```diff
@@ -93,6 +96,41 @@ public Response insert(RESTInputAdminRule inputAdminRule) throws NotFoundRestEx,
         }
     }
 
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED, value = "geofenceTransactionManager")
```

Comment:
I would annotate the service level method or even the dao one instead of the REST endpoint method. The controller layer should not take care of storage details like transaction management.